### ISSUE TITLE
Possible security concern - Drone instance misused for bitcoin mining

### ISSUE THREAD

**Issue by MauriceNino** [harness/drone#3307]
In the last few days, my VPS seemed to be attacked and used for bitcoin mining.

What I recognized happened:
- Some user got himself added to my users list in drone (at least just as a User, not Admin)
- He managed to start a drone container, running a script, without it showing in the UI
- The container was running [this repository](https://github.com/sirdapa/deepcc) (a bitcoin miner, presumably)

Also, I am not the only person having these issues lately, as it seems - [this forum poster here](https://community.harness.io/t/my-server-hacked-with-drone-io/13689) has the same exact problem, 6 days ago, which is also roughly when it started for me.

Did I misconfigure something, and it just happened to collide with another user's issues? Or is there actually a security concern within Drone?

This is my `docker-compose.yml`:

<details>
   <summary>File</summary>

```yml
version: "3.5"

services:
  drone_server:
    image: drone/drone:2
    restart: unless-stopped
    environment:
      - DRONE_TLS_AUTOCERT=false
      - DRONE_AGENTS_ENABLED=true
      - DRONE_GITHUB=true
      - DRONE_GITHUB_CLIENT_ID=${DRONE_GITHUB_CLIENT_ID}
      - DRONE_GITHUB_CLIENT_SECRET=${DRONE_GITHUB_CLIENT_SECRET}
      - DRONE_RPC_SECRET=${DRONE_RPC_SECRET}
      - DRONE_SERVER_HOST=drone.xxxx.com
      - DRONE_SERVER_PROTO=https
      - DRONE_LOGS_TRACE=true
      - DRONE_USER_CREATE=username:MauriceNino,admin:true
    volumes:
      - data:/data
    expose:
      - "80"
      - "443"
    networks:
      - proxy_net

  drone_runner:
    image: drone/drone-runner-docker:1
    restart: always
    depends_on:
      - drone_server
    environment:
      - DRONE_RPC_SECRET=${DRONE_RPC_SECRET}
      - DRONE_RPC_HOST=drone.xxxx.com
      - DRONE_RPC_PROTO=https
      - DRONE_RUNNER_CAPACITY=2
      - DRONE_RUNNER_NAME=LINUX_RUNNER
      - DRONE_LOGS_TRACE=true
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
    expose:
      - "3000"
    networks:
      - proxy_net

volumes:
  data:

networks:
  proxy_net:
    external: true
```
          
</details>

**Comment by donny-dont** [comment]
Did you set `DRONE_USER_FILTER`? 

See https://docs.drone.io/server/provider/github/

**Comment by MauriceNino** [comment]
No I did not, thanks for making me aware. Seems like that command was not there when I first set up Drone.

But even then, is a normal user allowed to register a repo and start CI runs that are invisible to the UI? Still seems like a huge concern to me.

**Comment by bradrydzewski** [comment]
You can limit registration by user or organization:
https://docs.drone.io/server/user/registration/

You can disable self-registration (only an admin can create a user account when you do this)
https://docs.drone.io/server/reference/drone-registration-closed/

These options are available to anyone using Drone 1.0.0 or higher.  If you are running an older version of Drone (0.8 or below) you may need other options.

**Comment by MauriceNino** [comment]
I am currently running the latest version - what I meant is that those options weren't present when I set up Drone about a year ago and seem to have been added after an update which I missed, therefore silently exposing my instance to wrongdoings like this.

> This is by design

I still feel like this is a concern and not how things _should_ be handled by default. But at least it's mentioned in the main install docs nowadays, unlike back then.
